### ISSUE TITLE
compatibility: typeset -u

### ISSUE THREAD

**Issue by bll123** [oilshell/oil#764]
compatibility with ksh, bash

I happened to notice that typeset -u is not implemented.
typeset -l is not either.

**Comment by andychu** [comment]
Do you have an example where this is used?  Generally we prioritize features to make specific programs run, e.g.

https://github.com/oilshell/oil/wiki/Shell-Programs-That-Run-Under-OSH

Also, I believe most usages should be rewriteable with `${x,,}` and so forth, which I think makes a lot more sense.

https://github.com/oilshell/oil/wiki/Language-Design-Principles

**Comment by bll123** [comment]
I use it, if available, as opposed to using 'tr'.  It is available in both ksh and bash.
Whereas ${x,,} is only a bash-ism.

https://github.com/search?p=20&q=%22typeset+-u%22&type=Code

Seems like some of the oracle scripts use it fairly often.
And occasionally in other scripts.

I agree it is rather low priority.

**Comment by andychu** [comment]
OK a lot of those usages are actually zsh `typeset -U` which means "unique entries" in an array.

But these other searches look like legit use:

https://github.com/search?q=%22declare+-u%22&type=Code

https://github.com/search?q=%22declare+-l%22&type=Code

https://github.com/search?p=4&q=%22typeset+-l%22&type=Code

I would take a patch for it, since it seems like we will have it eventually... The strategy / priority now is to implement the basic functionality in a fast and robust way, and then hopefully users will be motivated to contribute stuff like this.  There'ss a very long tail of features that aren't used that often in a relative sense, but ARE in an absolute sense (because there are so many shell users)

**Comment by bll123** [comment]
Yes, I wasn't counting the typeset -U; I waded through 30 pages of results.
I didn't see any "major" scripts.  But scripts like the oracle dbmon live forever.
